### DISCUSSION TITLE
search events from kafka topic by offset or range of offsets

### DISCUSSION THREAD

**Discussion by nitin1677** [provectus/kafka-ui#2720] [Q&A] [answered]
How can we search events from a kafka topic for given offset or range of offsets?

Variables bound to groovy context includes the following but missing offset: 
      partition, timestampMs, keyAsText, valueAsText, header, key (json if possible), value (json if possible).

**Comment by Haarolean** [comment, marked as answer]
Hey, you can select offsets while a smart filter is applied.
<img width="508" alt="image" src="https://user-images.githubusercontent.com/1494347/195052490-69aa123d-3fb6-4bf4-8a3d-880f9ab76d4e.png">

**Reply by nitin1677** [reply]
> Hey, you can select offsets while a smart filter is applied. <img alt="image" width="508" src="https://user-images.githubusercontent.com/1494347/195052490-69aa123d-3fb6-4bf4-8a3d-880f9ab76d4e.png">

Kafka UI Version: 10.10.2022 11:37:28 ([0939d61](https://github.com/provectus/kafka-ui/commit/0939d61))
Input:
Seek Type: Offset: 19124510
Partitions: All Items are selected.
Key Serde: String
Content Serde: String
Filter: keyAsText == "Commit Heartbeat"
Right most Drop Down selected as "Oldest First"

Result: Output starts from first offset 19124496 of the topic, instead of given offset 19124510 as seek type.

Expected Result: show only payload with given offset, instead all results from first offset, as right most drop down is selected as "Oldest First"

I am unable to add screenshot, so given all inputs as text above.

**Reply by Haarolean** [reply]
@nitin1677 okay, it might be related to #2624, we'll take a look

**Comment by iliax** [comment]
@nitin1677 unfortunately offsets positioning is working on master image now. As @Haarolean mentioned it is related to #2624 and will be fixed after that issue resolved. 

Btw, we will add "offset" var to smart filters (https://github.com/provectus/kafka-ui/pull/2732)
